internal/ingress/status/status.go

```diff
@@ -220,6 +239,11 @@ func (s *statusSync) runningAddresses() ([]v1.IngressLoadBalancerIngress, error)
 			continue
 		}
 
+		if pod.GetDeletionTimestamp() != nil {
+			klog.InfoS("POD is terminating", "pod", klog.KObj(&pod), "node", pod.Spec.NodeName)
+			continue
+		}
```

Comment:
Isn't the deletion timestamp set by the API server? So if a pod has finalizers or connections are still draining, I'm not sure if the controller is still running while already having the timestamp set.

Reply:
Yes, I'm aware of that. But setting `deletionTimestamp` does not mean the pod is actually terminating as the actual shutdown is initiated by `kubelet` at last and, in case of `ingress-nginx` might take longer due to the `preStopHook`. So the container itself is unready at that point, which means it's not eligible for new connections, but still running and able to take over leadership until it's getting asked to terminate after either the `preStopHook` completion or timeout (240s IIRC).

Reply:
So, usage example for here, where it can have shortlived requests, and current logic will not work.
* Amazon ELB - it awfully slow to update endpoints.
* DNS, where LoadBalancer status is several A records, so I want to wait for record TTL before killing the pod.

Same for K8s services, because of "eventual consistency" but update of IPTables/IPVS requires several seconds, so `sleep 15` everywhere makes internal networking work nice (of course it depends on the application, I'm talking about HTTP here).

I mean, this is not an Ingress nginx specific issue, this is how the k8s are designed from the ground up.

Termination handling eventually will be reimplemented everywhere, where pod discovery mimics service discovery logic.

The same issue was in External-dns for quite some time: https://github.com/kubernetes-sigs/external-dns/pull/3447#issuecomment-1453460602